File: eighteenthWeek/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = cur_profit = 0
        for i in range(len(prices)-1):
            cur_profit = max(0, cur_profit+prices[i+1]-prices[i])
            max_profit = max(max_profit, cur_profit)
        return max_profit

File: firstWeek/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null) {
            return 0;
        }
        
        int maxProfit = 0;
        int currentProfit = 0;
        for (int i = 1; i < prices.length; i++) {
            currentProfit = Math.max(currentProfit + prices[i] - prices[i-1], 0);
            maxProfit = Math.max(maxProfit, currentProfit);
        }
        return maxProfit;
    }
}
